catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -0,0 +1,206 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.plugin.defaultsecurity;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.codice.alliance.catalog.core.api.impl.types.SecurityAttributes;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.opensaml.core.xml.schema.XSString;
+import org.opensaml.saml.saml2.core.AttributeStatement;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.assertion.SecurityAssertion;
+
+public class DefaultSecurityAttributeValuesPlugin implements PreIngestPlugin {
+
+    private static final SecurityAttributes SECURITY_ATTRIBUTE_DESCRIPTORS =
+            new SecurityAttributes();
+
+    public static final String DEFAULTMARKINGS = "defaultMarkings";
+
+    private static Multimap<String, String> sysHighToMetacardAttributeMapping =
+            HashMultimap.create();
+
+    ddf.security.Subject getSystemSubject() {
+        return org.codice.ddf.security.common.Security.getInstance()
+                .getSystemSubject();
+    }
+
+    /**
+     * Retrieves the system high attributes and the mapping of the system attribute names to
+     * metacard security markings. Returns a hash map of the metacard markings
+     * to the value of its corresponding system attribute.
+     *
+     * @return Map of the metacard security markings to the value of their corresponding system high
+     * attribute.
+     */
+    private Map<String, Attribute> getHighwaterSecurityMarkings() {
+        Map<String, Attribute> securityMarkings = new HashMap<>();
+        ddf.security.Subject system =
+                org.codice.ddf.security.common.Security.runAsAdmin(() -> getSystemSubject());
+        SecurityAssertion assertion = system.getPrincipals()
+                .oneByType(SecurityAssertion.class);
+        List<AttributeStatement> attributeStatements = assertion.getAttributeStatements();
+        for (AttributeStatement curStatement : attributeStatements) {
```

Comment:
You can reduce cyclomatic complexity a little bit and improve overall performance with a couple of small tweaks. Take a look at this version of this block:
```java
        for (AttributeStatement curStatement : attributeStatements) {
            for (org.opensaml.saml.saml2.core.Attribute attribute : curStatement.getAttributes()) {
                Collection<String> attributeNames =
                        sysHighToMetacardAttributeMapping.get(attribute.getName());

                if (attributeNames.isEmpty()) {
                    continue;
                }
                HashSet<Serializable> attributeValues = attribute.getAttributeValues()
                        .stream()
                        .filter(curValue -> curValue instanceof XSString)
                        .map(XSString.class::cast)
                        .map(XSString::getValue)
                        .collect(Collectors.toCollection(HashSet::new));

                for (String attributeName : attributeNames) {
                    HashSet<Serializable> values = new HashSet<>(attributeValues);

                    securityMarkings.compute(attributeName, (k, v) -> {
                        if (v != null) {
                            values.addAll(v.getValues());
                        }
                        return new AttributeImpl(k,
                                (List<Serializable>) ImmutableList.copyOf(values));
                    });
                }
            }
        }
```